qupath-extension-bioformats/src/main/java/qupath/lib/images/servers/bioformats/BioFormatReaderPool.java

```diff
@@ -0,0 +1,86 @@
+package qupath.lib.images.servers.bioformats;
+
+import loci.formats.*;
+import loci.formats.ome.OMEPyramidStore;
+
+import java.util.Optional;
+
+/**
+ * <p>A {@link ReaderPool} to use with Bio-Format images.</p>
+ * <p>It uses the {@link BioFormatReaderWrapper}.</p>
+ */
+class BioFormatReaderPool extends ReaderPool {
```

Comment:
> Usually there should be a logger first as well - here, I think it would make sense to log at the debug level when createReaderWrapper is called.

Until now I've never used `logger.debug`, when should I use it?

Reply:
I'd say whenever there's info that could be useful to track down an error, but we wouldn't want logged routinely (since `info` level is the default).

Then `logger.trace` helps for *really* fine-grained information (e.g. it might be used to record rendering times in the viewer). Logs at `trace` level should almost never be turned on, because they will result in enormous logs.

Logging levels aren't used consistently enough in QuPath, but the new log viewer makes them far more useful.